### PR TITLE
capture request data with aiohttp multipart/form-data

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/32473

If both `files` and `data` are passed in the request, only what's in `files` gets added to aiohttp.FormData in the code below. `data` gets dropped and is not encoded into request.

https://github.com/Azure/azure-sdk-for-python/blob/2d0123c5e29ee84ef75cd5771320b75591dea729/sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py#L158-L175

For example, if I'm doing multipart/form-data with files and `data={"response_format": "srt"}`, I would expect my request body to include the response_format in the form-data like below:

```
b'--df13136170d9e03a0e82350af6b2afb7\r\nContent-Disposition: form-data; name="response_format"\r\n\r\nsrt\r\n--df13136170d9e03a0e82350af6b2afb7\r\nContent-Disposition: form-data; name="file"; filename="hello.m4a"\r\nContent-Type: application/octet-stream\r\n\r\n\x00\x00\x00\x18ftypmp42\x00\x00\x00\x00mp41isom\x00\x00\x00(uuid\\\xa7\x08\xfb2\x8eB\x05\xa8ae\x0e\xca\n\x95\x96\x00\x00\x00\x0c10.0.22621.0\x00\
```

In the current implementation, the request.data (and in this example, response_format) is ignored. This PR captures data and includes/uncomments tests for async.

### CODE REVIEW

**Review by xiangyan99 — APPROVED** [review]
LGTM

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def _get_request_data(self, request):
         :return: The request data
         """
         if request.files:
-            form_data = aiohttp.FormData()
+            form_data = aiohttp.FormData(request.data or {})
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:167]: Does this behaviour now match what happens in a sync request?

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:167]: requests library accepts both data and files in request method. aiohttp only accepts data.

Comment (kristapratico) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:167]: @xiangyan99 is right. requests accepts both files and data and handles building the multipart/form-data request body for us. In aiohttp, we must make sure we include both into the `data` for them to do the same.

**Review by lmazuel — APPROVED** [review]

**Comment by kristapratico** [comment]
core, containerregistry, and tables CI failures seem unrelated with the same tests failing on the main build.

**Pull request merged**